cllc-public-app/ClientApp/src/app/components/capacity-table/capacity-table.component.ts

```diff
@@ -0,0 +1,74 @@
+import { Component, Input, forwardRef } from '@angular/core';
+import { FormBuilder, FormArray, FormGroup, NG_VALUE_ACCESSOR } from '@angular/forms';
+import { ServiceArea } from '@models/service-area.model';
+import { BaseControlValueAccessor } from './BaseControlValueAccessor';
+
+
+@Component({
+  selector: 'app-capacity-table',
+  templateUrl: './capacity-table.component.html',
+  styleUrls: ['./capacity-table.component.scss'],
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => CapacityTableComponent),
+      multi: true
+    }
+  ]
+})
+export class CapacityTableComponent extends BaseControlValueAccessor<ServiceArea[]> {
+    @Input() isIndoor: boolean;
+
+    form: FormArray;
+    registerOnChange(fn: any) { this.onChange = fn; }
+    registerOnTouched(fn: any) { this.onTouched = fn; }
+
+    constructor(private fb: FormBuilder) {
+        super();
+
+        this.form = fb.array([]);
+
+        this.form.valueChanges.subscribe(val => {
+            this.onChange(val);
+            this.value = val;
+        });
+    }
+
+    writeValue(serviceAreas: ServiceArea[]) {
+        if (serviceAreas) {
+            super.writeValue(serviceAreas);
+            // this sucks, maybe there's a better way, just trying to
+            // set the value of the array to the new value
```

Comment:
@GeorgeWalker and @msehudi somewhere in here is causing me a ton of grief. I want to set the form array to the new values however this method has console errors that are thrown complaining the form controls don't work. I have tried a few things and I've seen some comments on stackoverflow saying to use a timeout which I won't do. Any suggestions?

Reply:
Hey @brianorwhatever, try placing the formarray inside a from group. I suspect the html binding "[formGroup]='form'" might be the cause of the unexpected behavior. The binding expects a FormGroup instead of a FormArray

Reply:
hmm I bet you are right. I originally had it that way but wanted to remove that group layer so that the backend would be simpler but I think you are right and I should revert back to that. Thanks!

Reply:
Also you might need to cast the array elements - see my "FixChildren" routine for the change log entries for an example.